esiclient/v1/switch.py

```diff
@@ -124,6 +124,32 @@ def take_action(self, parsed_args):
         return ["Port", "VLANs"], data
 
 
+class ListSwitches(command.Lister):
+    """List Switches """
+
+    log = logging.getLogger(__name__ + ".ListSwitches")
+
+    def get_parser(self, prog_name):
+        parser = super(ListSwitches, self).get_parser(prog_name)
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        ironic_client = self.app.client_manager.baremetal
+
+        ports = list((port for port in ironic_client.port.list(detail=True)))
```

Comment:
Would ``ports = ironic_client.port.list(detail=True)`` also work? It feels like you are iterating through a list in order to create a list.

Reply:
yes totally agree and it worked!